src/Intercom/Clients/AdminsClient.cs

```diff
@@ -64,12 +64,12 @@ public Admin View (String id)
         public Admin View(Admin admin)
         {
             if (admin == null) {
-                throw new ArgumentNullException ("'admin' argument is null.");
+                throw new ArgumentNullException (nameof(admin));
             }
 
             if (String.IsNullOrEmpty(admin.id))
             {
-                throw new ArgumentNullException ("you must provide value for 'admin.id'.");
+                throw new ArgumentException ("you must provide value for 'admin.id'.");
```

Comment:
Hi!  No, I don't think so.  In general, the best practice for `ArgumentNullException`'s are to pass in the name of the variable that is `null`.  If there is something else wrong with it, then I believe a different type of exception is supposed to be raised like `ArgumentException`.  I could be totally wrong though 😄